### PR TITLE
Dev add tests recruitment population

### PR DESCRIPTION
**Pull request by nathanvaughan-NOAA**
# What is the feature?
* Update recruitment and population Rcpp interface test files to the new standard format

# How have you implemented the solution?
* Used the template function to create new files and move over tests

# Does the PR impact any other area of the project, maybe another repo?
* I don't think so

### CODE REVIEW

File: tests/testthat/test-rcpp-population-interface.R [hunk 1]
```diff
@@ -1,39 +1,140 @@
...
+#' This file follows the format generated by FIMS:::use_testthat_template().
+#' Necessary tests include input and output (IO) correctness [IO
+#' correctness], edge-case handling [Edge handling], and built-in errors and
+#' warnings [Error handling]. See `?FIMS:::use_testthat_template` for more
+#' information. Every test should have a @description tag that takes up just
+#' one line, which will be used in the bookdown report of {testthat} results.
+
+# test_rcpp_population_interface ----
+## IO correctness ----
+test_that("test_rcpp_population_interface() works with correct inputs", {
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-rcpp-population-interface.R:11]: Non-blocking thought: Technically, it would be good to test population.evaluate() as well, though I'm not entirely sure what input it requires. I'm leaning toward merging this PR first, and I can look into whether we can add tests for population.evaluate() afterward. Let me know if you have any suggestions.

File: tests/testthat/test-rcpp-population-interface.R [hunk 2]
```diff
@@ -1,39 +1,140 @@
...
+  population$nyears$set(nyears)
+
+
+  #' @description Test that the population id is 1.
+  expect_equal(
+    object = population$get_id(),
+    expected = 1
+  )
+
+  #' @description Test that the log_M values are all -1.
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-rcpp-population-interface.R:45]: Could you move line 45 so that it comes just before the `expect_equal()` call on line 48?

File: tests/testthat/test-rcpp-population-interface.R [hunk 3]
```diff
@@ -1,39 +1,140 @@
...
+  #' @description Test that the log_M values are not estimated.
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-rcpp-population-interface.R:46]: Could you move line 46 so that it comes just before the `expect_equal()` call on line 52?

The suggestion also applies to the remaining #' @description lines below.

File: tests/testthat/test-rcpp-population-interface.R [hunk 4]
```diff
@@ -1,39 +1,140 @@
...
+    expect_false(
+      object = population$log_init_naa[i]$estimated
+    )
+  }
+
+  clear()
+})
+
+## Edge handling ----
+test_that("test_rcpp_population_interface() returns correct outputs for edge cases", {
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-rcpp-population-interface.R:99]: Feel free to remove lines 99 through 104 and include a comment saying no edge case tests for now.

File: tests/testthat/test-rcpp-population-interface.R [hunk 5]
```diff
@@ -1,39 +1,140 @@
...
+  # @description Test that test_rcpp_population_interface(x) returns an error.
+  #expect_error(
+  #  object = test_rcpp_population_interface(x)
+  #)
+})
+
+## Error handling ----
+test_that("test_rcpp_population_interface() returns correct error messages", {
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-rcpp-population-interface.R:107]: It looks like there aren’t any actual tests here. Maybe remove lines 107–140 and add a comment like 'No built-in errors or warnings to test'.

**Review by Bai-Li-NOAA — APPROVED** [review]
Thank you, @nathanvaughan-NOAA for refactoring our tests! The PR looks great. I only have a few minor formatting suggestions. Once you’ve made those changes, please let me know and **I’ll merge the commits into `dev-refactor-rcpp-tests` instead of `dev`**.

I also agree that the missing coverage in `add_to_fims_tmb_internal()`, `finalize()`, and the JSON-related sections should be addressed through integration tests, so no further action is needed in this PR.

File: tests/testthat/test-rcpp-population-interface.R (hunk 1 above)
Comment (nathanvaughan-NOAA) [reply tests/testthat/test-rcpp-population-interface.R:11]: I agree, and it annoyed me too, but population.evaluate() runs the entire population loop which requires everything to be initialized so I assume this can only be tested as part of the integration tests.

File: tests/testthat/test-rcpp-recruitment-interface.R [hunk 6]
```diff
@@ -14,26 +23,91 @@ test_that("Recruitment input settings work as expected", {
...
+    )
+  }
+
+  #expected_lpdf <- sum(log(stats::dnorm(log_devs, 0, 0.7)))
+  clear()
+})
 
-  expected_lpdf <- sum(log(stats::dnorm(log_devs, 0, 0.7)))
+## Edge handling ----
+test_that("test_rcpp_recruitment_interface() returns correct outputs for edge cases", {
```
[the remark this answers is not in the document]
Comment (nathanvaughan-NOAA) [reply tests/testthat/test-rcpp-recruitment-interface.R:83]: Will do, the only reason I commented out rather than deleted the unneeded sections was so that we had the correct template format there later rather than having to go look for it in other tests.

**Comment by nathanvaughan-NOAA** [comment]
Thanks @Bai-Li-NOAA, I just pushed those changes. We should be good to go once the checks pass.

File: tests/testthat/test-rcpp-population-interface.R (hunk 2 above)
Comment (nathanvaughan-NOAA) [reply tests/testthat/test-rcpp-population-interface.R]: done

File: tests/testthat/test-rcpp-population-interface.R (hunk 3 above)
Comment (nathanvaughan-NOAA) [reply tests/testthat/test-rcpp-population-interface.R]: done

File: tests/testthat/test-rcpp-population-interface.R (hunk 4 above)
Comment (nathanvaughan-NOAA) [reply tests/testthat/test-rcpp-population-interface.R]: done

File: tests/testthat/test-rcpp-population-interface.R (hunk 5 above)
Comment (nathanvaughan-NOAA) [reply tests/testthat/test-rcpp-population-interface.R]: done
